src/interfaces/token.ts

```diff
@@ -59,39 +70,64 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
   }
 
   /**
-   * Fetch token balances from the {@link TokenInterface.supported} list
-   * for a particular address.
-   * @param address
+   * Fetch token balance for a particular account and token addresses.
+   *
+   * @param account user wallet address
+   * @param tokenAddresses list of token addresses
+   *
+   * @returns list of balances for the supported tokens
    */
-  async balances(address: Address): Promise<Balance[]> {
-    const vaultBalances = await this.yearn.vaults
-      .balances(address)
-      .then(balances => balances.filter(token => token.balance !== "0"));
-
-    switch (this.chainId) {
-      case 1:
-      case 1337: {
-        let zapperBalances: Balance[] = [];
-        try {
-          zapperBalances = await this.yearn.services.zapper.balances(address);
-        } catch (error) {
-          console.error(error);
-        }
-        const vaultBalanceAddresses = new Set(vaultBalances.map(balance => balance.address));
-        zapperBalances = zapperBalances.filter(balance => !vaultBalanceAddresses.has(balance.address));
-        return zapperBalances.concat(vaultBalances);
+  async balances(account: Address, tokenAddresses?: Address[]): Promise<Balance[]> {
```

Comment:
I get that the new argument name makes more sense, but we don't want to introduce a breaking change if people are using labelled arguments

Reply:
It's not a breaking change, you just need to give it a string as the first argument, i.e. you don't give it any name when calling it

```typescript
function foo(bar: string): string {
  return bar;
}

foo("baz") // baz
```

with named params it would be a breaking change because you do need to specify which named param you want to give the argument to

```typescript
function foo(bar: { baz: string }): string {
  return bar.baz;
}

foo({ baz: "string" }); // string

```

Reply:
Yeah we should definitely switch to named parameters, much more expressive. 

But what happens if an external consumer of the sdk Is currently using sdk.token.balances({address: "0x..."})? Won't the next version break that because you now need to have sdk.token.balances({account: "0x..."})?

Reply:
They currently dont have that option of using the interface you describe, users cant pass objects as an argument as of now.

Reply:
Yeah, you can only give it a string as an argument as is